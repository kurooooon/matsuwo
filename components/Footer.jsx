import React from 'react';
import styled from "@emotion/styled";
import { FaInstagram, FaFacebook, FaYoutube, FaTwitter } from "react-icons/fa";
import { FiMail } from "react-icons/fi";
import TickTok from '../static/images/tiktok.svg'

const FooterSection = styled.section`
  padding: 6rem 0 4rem 0;

  @media (min-width: 1140px) {
    padding: 6em 0 6em 0;
  }

  @media (max-width: 980px) {
    padding: 5rem 3rem 3rem 3rem;
  }

  @media (max-width: 736px) {
    padding: 3rem 1.5rem 3rem 1.5rem;
  }

  @media (min-width: 480px) {
    padding: 5rem 3rem 5rem 3rem;
  }

  background-color: #4686a0;
  color: rgba(255, 255, 255, 0.75);
  background-attachment: fixed, fixed, fixed;
  background-position: top left, center center, center center;
  background-size: auto, cover, cover;
  text-align: center;
  background-image: url("https://res.cloudinary.com/kurooooon/image/upload/v1579370033/matsuwo/overlay2_xpyeat.png"), url("https://res.cloudinary.com/kurooooon/image/upload/v1579370033/matsuwo/overlay4_fyhlai.svg"), linear-gradient(45deg, #614d20, #333 95%);

  a {
    border-bottom-color: rgba(255, 255, 255, 0.5);
  }

    a:hover {
      color: #ffffff;
    }

  h1, h2, h3, h4, h5, h6, strong, b {
    color: #ffffff;
  }

  header p {
    color: #ffffff;
  }

  header.major:after {
    background: #ffffff;
  }

  input[type="submit"],
  input[type="reset"],
  input[type="button"],
  button,
  .button {
    box-shadow: inset 0 0 0 1px #ffffff;
    color: #ffffff !important;
  }

  input[type="submit"]:hover,
  input[type="reset"]:hover,
  input[type="button"]:hover,
  button:hover,
  .button:hover {
    background-color: rgba(255, 255, 255, 0.125);
  }

  input[type="submit"]:active,
  input[type="reset"]:active,
  input[type="button"]:active,
  button:active,
  .button:active {
    background-color: rgba(255, 255, 255, 0.25);
  }

  input[type="submit"].primary,
  input[type="reset"].primary,
  input[type="button"].primary,
  button.primary,
  .button.primary {
    background-color: #ffffff;
    box-shadow: inset 0 0 0 1px #ffffff !important;
    color: #4686a0 !important;
  }

  input[type="submit"].primary:hover,
  input[type="reset"].primary:hover,
  input[type="button"].primary:hover,
  button.primary:hover,
  .button.primary:hover {
    background-color: rgba(255, 255, 255, 0.125) !important;
    color: #ffffff !important;
  }

  input[type="submit"].primary:active,
  input[type="reset"].primary:active,
  input[type="button"].primary:active,
  button.primary:active,
  .button.primary:active {
    background-color: rgba(255, 255, 255, 0.25) !important;
  }

  ul.major-icons li .icon {
    border-color: #ffffff;
  }

  .icon.major {
    color: #ffffff;
  }

  .icons {
    margin: 0;
  }

  .copyright {
    font-size: 0.8rem;
    list-style: none;
    margin: 2rem 0 0 0;
    padding: 0;
  }

  .copyright li {
    border-left: solid 1px;
    display: inline-block;
    line-height: 1rem;
    margin-left: 1rem;
    padding: 0 0 0 1rem;
  }

  .copyright li:first-child {
    border-left: 0;
    margin-left: 0;
    padding: 0;
  }
`;

const FooterLink = styled.a`
  font-size: 1rem;
  border-bottom: none;

  @media (min-width: 480px) {
    font-size: 1.1rem;
  }

  @media (min-width: 1140px) {
    font-size: 1.2rem;
  }
`;

const CopyRight = styled.ul`
  li {
    font-size: 0.8rem;
    border: 0;
    display: block;
    padding: 0;
    line-height: 1rem;

    @media (min-width: 480px) {
      display: inline-block;
    }

    :last-child {
      margin: 0.8rem 0 0 0;

      @media (min-width: 480px) {
        border-left: solid 1px;
        margin-left: 1em;
        padding: 0 0 0 1em;
      }
    }
  }
`;

const FooterList = styled.ul`
  cursor: default;
  list-style: none;
  padding-left: 0;
  margin-bottom: 1.2rem;

  @media (min-width: 480px) {
    margin-bottom: 2rem;
  }

  li {
    display: inline-block;
    padding: 0 1.25rem 0 0;
  }

  li:last-child {
    padding-right: 0;
  }

  li .icon:before {
    font-size: 1.5rem;
  }
`;

const Footer = function Presenter ({ id }) {
  return (
    <FooterSection>
      <FooterList>
        <li>
          <FooterLink
            href="https://www.youtube.com/channel/UCRSPD9OHzBjDfY9EFE4hDHw"
            target="_blank"
            aria-label="ToLo Records Channel / Youtube"
          >
            <FaYoutube alt="ToLo Records Channel / Youtube" />
          </FooterLink>
        </li>
        <li>
          <FooterLink
            href="https://twitter.com/MATUWOMATUWO"
            target="_blank"
            aria-label="@MATUWOMATUWO / Twitter"
          >
            <FaTwitter alt="@MATUWOMATUWO / Twitter" />
          </FooterLink>
        </li>
        <li>
          <FooterLink
            href="https://www.facebook.com/ryuhei.matsuo.50"
            target="_blank"
            aria-label="Ryuhei Matuso / Facebook"
          >
            <FaFacebook alt="Ryuhei Matuso / Facebook" />
          </FooterLink>
        </li>
        <li>
          <FooterLink
            href="https://www.instagram.com/ryuhei_matsuo/"
            target="_blank"
            aria-label="ryuhei_matsuo / instagram"
          >
            <FaInstagram alt="ryuhei_matsuo / instagram" />
          </FooterLink>
        </li>
        <li>
          <FooterLink
            href="https://www.tiktok.com/@matumatumatu"
            target="_blank"
            aria-label="@matumatumatu / tiktok"
          >
            <TickTok width="16" alt="@matumatumatu / tiktok" />
          </FooterLink>
        </li>
        <li>
          <FooterLink
            href="mailto:matsuwo611@gmail.com"
            target="_blank"
            aria-label="mail to matsuwo"
          >
            <FiMail alt="mail to matsuwo" />
          </FooterLink>
        </li>
      </FooterList>
      <CopyRight>
        <li>&copy; matsuwo / 松尾竜平</li>
        <li>
          Design: <a href="https://html5up.net/">HTML5 UP</a>
        </li>
      </CopyRight>
    </FooterSection>
  )
}

export default Footer;