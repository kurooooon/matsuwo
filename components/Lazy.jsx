import React from 'react';
import { InView } from 'react-intersection-observer'

const Lazy = function Lazy({ children, rootMargin, triggerOnce }) { 
  return (
    <InView triggerOnce={triggerOnce} rootMargin={rootMargin}>
      {({ inView, ref }) => (
        <div ref={ref}>{inView ? children : null}</div>
      )}
    </InView>
  );
}

export default Lazy