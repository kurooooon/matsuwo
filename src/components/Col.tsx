import styled from '@emotion/styled';

export const Col = styled.div`
  @media (min-width: 1140px) {
    width: 50%;
  }
`;
