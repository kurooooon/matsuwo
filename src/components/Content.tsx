import { css } from '@emotion/react';
import { HTMLAttributes, ReactNode } from 'react';

interface Props extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
}

export const Content = ({ children, ...rest }: Props) => {
  return (
    <div
      css={css`
        max-width: 1000px;
        margin: 0 auto;
      `}
      {...rest}
    >
      {children}
    </div>
  );
};
