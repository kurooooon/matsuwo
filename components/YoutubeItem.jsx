import React from 'react';
import YouTube from 'react-youtube';
import styled from '@emotion/styled';
import Lazy from './Lazy';

const Title = styled.p`
  text-align: center;
  margin-bottom: 15px;
  font-size: 1.1rem;
`;

const YoutubeItem = React.memo(function Presenter ({ id, title }) {
  return (
    <div>
      <Lazy triggerOnce rootMargin='100px 0px'>
        <YouTube opts={{width: '100%'}} videoId={id} />
      </Lazy>
      <Title>{title}</Title>
    </div>
  );
}, (prevProps, nextProps) =>
  prevProps.id === nextProps.id &&
  prevProps.title === nextProps.title
)

export default YoutubeItem;