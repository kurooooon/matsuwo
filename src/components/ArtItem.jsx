import React, { useCallback } from 'react';
import { Image } from "cloudinary-react";
import styled from '@emotion/styled';
import Lazy from './Lazy';

const Art = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const ArtTitle = styled.p`
  text-align: center;
  margin-top: 5px;
  margin-bottom: 15px;
  font-size: 1.1rem;
  width: 100%;
`;

const ArtLink = styled.a`
  border: none;
`;

export const ArtItem = React.memo(function Presenter ({ id, title, url }) {
  const onClick = useCallback(() => {
    window.dataLayer.push({
      event: 'clickArt',
      id,
      title
    }, [id, title, url])
  });
  return (
    <Art key={id}>
      <ArtLink onClick={onClick} href={url} target="_blank">
        <Lazy triggerOnce rootMargin='100px 0px'>
          <Image
            cloudName="kurooooon"
            publicId={`matsuwo/art/${id}`}
            width="100%"
            secure={true}
            alt=""
          />
        </Lazy>
        <ArtTitle>{title}</ArtTitle>
      </ArtLink>
    </Art>
  );
}, (prevProps, nextProps) =>
  prevProps.id === nextProps.id &&
  prevProps.title === nextProps.title &&
  prevProps.url === nextProps.url
);
