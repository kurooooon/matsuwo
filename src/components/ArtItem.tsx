import React, { useCallback } from 'react';
import { Image } from 'cloudinary-react';
import styled from '@emotion/styled';
import Lazy from './Lazy';
import { useTrackGA } from '../hooks/useTrackGA';

const Art = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const ArtTitle = styled.p`
  text-align: center;
  margin-top: 5px;
  margin-bottom: 15px;
  font-size: 1.1rem;
  width: 100%;
`;

const ArtLink = styled.a`
  border: none;
`;

type Props = {
  id: string;
  title: string;
  url: string;
};

const ArtItem = ({ id, title, url }: Props) => {
  const trackGA = useTrackGA();
  const onClick = useCallback(() => {
    trackGA({
      event: 'clickArt',
      id,
      title,
    });
  }, [id, title, url]);
  return (
    <Art key={id}>
      <ArtLink onClick={onClick} href={url} target="_blank">
        <Lazy triggerOnce rootMargin="100px 0px">
          <Image
            cloudName="kurooooon"
            publicId={`matsuwo/art/${id}`}
            width="100%"
            secure={true}
            alt=""
          />
        </Lazy>
        <ArtTitle>{title}</ArtTitle>
      </ArtLink>
    </Art>
  );
};

const areEquals = (prev: Props, next: Props) =>
  prev.id === next.id && prev.title === next.title && prev.url === next.url;

export default React.memo(ArtItem, areEquals);
