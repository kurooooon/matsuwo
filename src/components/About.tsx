import React from 'react';
import styled from '@emotion/styled';
import { SectionHeader } from './SectionHeader';
import { Row } from './Row';
import { Col } from './Col';
import Lazy from './Lazy';
import { SectionCss } from './Section';
import { Image } from 'cloudinary-react';

const AboutWrapper = styled.section`
  ${SectionCss}
  position: relative;
  background-color: #333;
  color: rgba(255, 255, 255, 0.75);
  background-attachment: fixed, fixed;
  background-image: url('https://res.cloudinary.com/kurooooon/image/upload/f_auto/v1579369001/matsuwo/back_wqixtg.jpg');
  z-index: 0;
  overflow: hidden;
  background-color: #000000dd;
  background-repeat: no-repeat;
  background-position: top;
  background-size: cover;

  @media (max-width: 1140px) and (-webkit-min-device-pixel-ratio: 0) {
    background-attachment: scroll;
  }

  ::before {
    content: '';
    position: absolute;
    top: -10px;
    bottom: -10px;
    left: -10px;
    right: -10px;
    background: inherit;
    filter: blur(10px);
    z-index: -1;
  }

  a {
    border-bottom-color: rgba(255, 255, 255, 0.5);
  }

  a:hover {
    color: #ffffff;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  strong,
  b {
    color: #ffffff;
  }

  header p {
    color: #ffffff;
  }

  input[type='submit'],
  input[type='reset'],
  input[type='button'],
  button,
  .button {
    box-shadow: inset 0 0 0 1px #ffffff;
    color: #ffffff !important;
  }

  input[type='submit']:hover,
  input[type='reset']:hover,
  input[type='button']:hover,
  button:hover,
  .button:hover {
    background-color: rgba(255, 255, 255, 0.125);
  }

  input[type='submit']:active,
  input[type='reset']:active,
  input[type='button']:active,
  button:active,
  .button:active {
    background-color: rgba(255, 255, 255, 0.25);
  }

  input[type='submit'].primary,
  input[type='reset'].primary,
  input[type='button'].primary,
  button.primary,
  .button.primary {
    background-color: #ffffff;
    box-shadow: inset 0 0 0 1px #ffffff !important;
    color: #333 !important;
  }

  input[type='submit'].primary:hover,
  input[type='reset'].primary:hover,
  input[type='button'].primary:hover,
  button.primary:hover,
  .button.primary:hover {
    background-color: rgba(255, 255, 255, 0.125) !important;
    color: #ffffff !important;
  }

  input[type='submit'].primary:active,
  input[type='reset'].primary:active,
  input[type='button'].primary:active,
  button.primary:active,
  .button.primary:active {
    background-color: rgba(255, 255, 255, 0.25) !important;
  }

  ul.major-icons li .icon {
    border-color: #ffffff;
  }

  .icon.major {
    color: #ffffff;
  }
`;

const AboutContainer = styled.div`
  @media (min-width: 1140px) {
    margin: 0 auto;
    max-width: calc(100% - 4em);
    width: 60em;
  }
`;

const AboutInnerWrapper = styled.div`
  background-color: #00000099;
  padding: 20px;
`;

const AboutDescription = styled.div`
  font-size: 1rem;

  @media (min-width: 480px) {
    font-size: 1.1rem;
  }

  @media (min-width: 1140px) {
    width: 50%;
    padding-left: 3rem;
  }
`;

const ProfileImageWrapper = styled.div`
  display: flex;
  margin-bottom: 2rem;
`;

const About = () => (
  <AboutWrapper>
    <AboutContainer>
      <AboutInnerWrapper>
        <SectionHeader inverse>
          <h2>About</h2>
        </SectionHeader>
        <Row>
          <Col>
            <ProfileImageWrapper>
              <Lazy triggerOnce rootMargin="100px 0px">
                <Image
                  cloudName="kurooooon"
                  publicId="matsuwo/profile_wkgtbv"
                  width="100%"
                  height="100%"
                  alt=""
                  secure={true}
                />
              </Lazy>
            </ProfileImageWrapper>
          </Col>
          <AboutDescription>
            <p>matsuwo / 松尾竜平</p>
            <p>
              北海道札幌での
              <a
                href="https://www.saatchiart.com/account/profile/218701"
                target="_blank"
                rel="noreferrer"
              >
                アート
              </a>
              の制作や個展、音楽制作やライブ活動を精力的に行い、2019年春より東京での活動をスタート。
              2013年matsuwo名義で
              <a
                href="https://tower.jp/artist/2149889/matsuwo"
                target="_blank"
                rel="noreferrer"
              >
                宅録CD
              </a>
              をリリース。
            </p>
            <p>
              歌詞と曲は心揺さぶられるできごと作品や言葉などありふれた日常の中からインスパイアされ主にJ-popの曲をlogic
              proやProtoolsで制作しており、音はいたってシンプルで分厚く重ねない。
              影響されたアーティストはゆず、ミスターチルドレン、サザンオールスターズ、宇多田ヒカルなどなど。
              ライブでは主にアコースティックギターの弾き語り形式で演奏。
              使用アコースティックギターはリサイクルショップで購入した1560円のモーリス。
              今まで使ってきたどれよりも弾きやすい。
            </p>
            <p>
              アートでは油絵を制作してイタリア、フランス、ニューヨークの展覧会やアートインレジデンスを経験後、音楽ジャケットやチラシのアートワークを担当したりアートフェアや個展などで絵画を発表している。
            </p>
          </AboutDescription>
        </Row>
      </AboutInnerWrapper>
    </AboutContainer>
  </AboutWrapper>
);

export default About;
