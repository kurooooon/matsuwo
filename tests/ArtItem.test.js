import React from 'react';
import { ArtItem } from '../components/ArtItem';
import renderer from 'react-test-renderer';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils'

test('<ArtItem> snapshot testing', () => {
  const component = renderer.create(
    <ArtItem
      id="4274559-XBAHEJFZ-7_tt3p6x"
      title="After the party"
      url="https://www.saatchiart.com/art/Painting-After-the-party/218701/4519690/view"
    />
  );
  mockAllIsIntersecting(true);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});