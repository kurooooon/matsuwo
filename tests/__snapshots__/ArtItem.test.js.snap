// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtItem> snapshot testing 1`] = `
<div
  className="css-rtde4j"
>
  <a
    className="css-jsj4j0"
    href="https://www.saatchiart.com/art/Painting-After-the-party/218701/4519690/view"
    onClick={[Function]}
    target="_blank"
  >
    <div />
    <p
      className="css-52lrx0"
    >
      After the party
    </p>
  </a>
</div>
`;
